### DISCUSSION TITLE
Best guide for explaining how data in a new document library column is shown in search results

### DISCUSSION THREAD

**Discussion by steveh2023** [microsoft-search/pnp-modern-search#3511] [Q&A]
Hi,

I have looked at most of the documentation on this site but I must have missed the info that explains how to surface data in a new document library column - can someone please tell me the best explanation to do this?

I have mapped the internal column name to a new RefinableString in managed properties - i have reindexed - i added a new layout slot then added it as a new column using manage columns and the data does not show in the search results...

I am surprised there are so many steps and I am not sure what i did is not missing a step or 2 - so where is the info on how to do this simple end user task please? I dont see any end to end guides on how to do this - I thought it the info might be here: https://microsoft-search.github.io/pnp-modern-search/scenarios/ - maybe I missed it, maybe this scenario above is not there...thanks.

**Comment by kasperbolarsen** [comment]
Selecting the correct Information Architecture and setting up the search mapping afterwards IS complicated, and some of us is making a living by knowing this stuff :-) Unfortunately I doubt there are any end-to-end guide covering your specific use case as each case always comes with a twist.
I can recommend the resources and courses provided by [Search ](https://searchexplained.com/).

**Reply by steveh2023** [reply]
I did the config correctly - the issue was the crawler was very very very slow because this site has literally hundreds of thousands of files....most of which are not needed - so we deleted them - then the crawler ran in reasonable time and the info appeared in the search results - so in this case volume of site data was the problem....thanks for the link to the https://searchexplained.com/ site but we don't have a budget for pricey training courses - sadly...

**Comment by gretchunkim** [comment]
Sometimes you have to let the indexing/crawling to complete its cycle. Try to take a look at the result tomorrow (12-16hrs) and see if you can find these values.

**Reply by steveh2023** [reply]
Yes it is ok now - thanks.

**Reply by steveh2023** [reply]
Yes it is ok now - thanks.

**Comment by wobba** [comment]
This is a good article as well https://learn.microsoft.com/en-us/microsoft-365/community/how-do-site-columns-become-managed-properties-thus-available-for-search. I'll shamelessly plug my book https://www.amazon.com/Troubleshooting-Search-SharePoint-Online-solving/dp/1080506411 which also covers columns->crawled properties->managed properties. And also site columns which may make life easier for many scenarios.

**Reply by steveh2023** [reply]
Thanks for the links - I might get your book.

**Reply by steveh2023** [reply]
I was reading the site columns link very briefly the other day but the column i was working with is just a normal new one on a document library ie not a site column and it was not working but as mentioned above in this case the config was correct but due to thousands and thousands of files the crawl was much slower than usual - these files were not needed - they were deleted - then the info appeared soon after....

**Reply by kasperbolarsen** [reply]
Search often depends on a proper Information Architecture in order to provide the requested outcome. I might be rather old school but so far my approach haven't failed me yet :-) 
The first question I ask myself is, "is there a risk this column can be of use on another list or library"? If there is even the slightest risk then the column should be created as a Site Column.
The second question is, "is there a risk that this Site Column can be of use on another Site Collection"? If so, then the Site Column should be created in the Content Type hub.
And in generel , always use a custom Content Type :-)

**Reply by steveh2023** [reply]
Thanks for the info especially:

 "is there a risk this column can be of use on another list or library"? If there is even the slightest risk then the column should be created as a Site Column."

I made this column as a new one to the document library but it should of been a site column after reading that I think.
